javascript/ql/lib/semmle/javascript/internal/ConceptsShared.qll

```diff
@@ -78,7 +78,7 @@ module Cryptography {
        * This may have no result - for example if the `CryptographicAlgorithm` used
        * is a stream cipher rather than a block cipher.
        */
-      abstract BlockMode getBlockMode();
+      BlockMode getBlockMode() { none() }
```

Comment:
As mentioned in other comments, it's possible to do this change of making it non-abstract for JS only.

My default stance on this is that it's nicer if the predicates you're supposed to implement are abstract, since that means you won't forget to do it. However, there is also part of me that is annoyed at having to implement `getBlockMode` for hashing cryptographic operations...

So I guess now is a good time to think about whether we _want_ to change this to have a default of `none()` for all languages or not. What are your own thoughts on this?

Reply:
sounds good to me :+1: